### PR TITLE
Added customer to projects create

### PR DESCRIPTION
**Pull request by adognameddog**
Associated PR on core: https://github.com/teamleadercrm/core/pull/5604
Associated PR on api-tests: https://github.com/teamleadercrm/api-tests/pull/279

I've altered the documentation for `projects.create` to include the optional customer.

### CODE REVIEW

**Review by Jdww — COMMENTED** [review]
Looks good but you need to build/regenerate the apiary.apib file 
https://github.com/teamleadercrm/api/blob/master/CONTRIBUTING.md

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /node_modules
 package-lock.json
 apiary.html
+/.idea
```
Comment (JordyDeGroeve) [inline .gitignore]: PP: not sure if we should add IDE specific ignores in here.

(feel free to ignore this)

**Review by JordyDeGroeve — DISMISSED** [review]

File: .gitignore (hunk 1 above)
Comment (VerstraeteBert) [reply .gitignore]: Here's a previous discussion on this;
https://bit.ly/2Izrffj

**Review by VerstraeteBert — COMMENTED** [review]
Here's a previous discussion on this; You can add these to your own global gitignore file.
https://bit.ly/2Izrffj

File: src/07-projects/projects.apib [hunk 2]
```diff
@@ -126,6 +126,12 @@ Create a new project.
                         + decision_maker
                         + member
                     + Default: `member`
+        + customer (object, optional)
+            + type (enum[string])
```
Comment (dietrichm) [inline src/07-projects/projects.apib]: PP: add `required` to the `type` definition (same for `id`). It is required whenever the customer is provided.

File: .gitignore (hunk 1 above)
Comment (dietrichm) [reply .gitignore]: We shouldn't indeed.

Comment (adognameddog) [reply .gitignore]: I fixed that. Much better now.

 :ok: by @JordyDeGroeve on 03/07/2018 16:29:47

**Review by JordyDeGroeve — APPROVED** [review]

**Review by rathesDot — APPROVED** [review]
Just wondering: Don't we provide a default for enums?

**Comment by adognameddog** [comment]
@rathesDot No, when providing a customer, both type and id are required. In the database however, "contact" is the default. Other implementations of "customer" in the api also require type and id.

**Comment by rathesDot** [comment]
Thanks for the explanation, @adognameddog! Makes sense!

**Review by psrpinto — APPROVED** [review]

**Review by dietrichm — APPROVED** [review]

**Review by steftrenson — APPROVED** [review]

**Pull request merged**